ssh-server/main.go

```diff
@@ -111,40 +116,78 @@ func serviceChannel(newChannel ssh.NewChannel) error {
 	if err != nil {
 		return fmt.Errorf("accepting channel: %v", err)
 	}
-	defer channel.Close()
 
-	// Start the command with a pseudo-terminal.
+	allow := map[string]bool{"shell": true, "exec": true, "pty-req": true, "window-change": true}
+	resize := make(chan *pty.Winsize, 1)
+	done := make(chan struct{})
+	for {
+		select {
+		case <-done:
+			return nil
+		case req, ok := <-requests:
+			if !ok {
+				return nil
+			}
+			if req.WantReply {
+				req.Reply(allow[req.Type], nil)
+			}
+			switch req.Type {
+			case "pty-req":
+				go func() {
+					newPTY(channel, resize)
+					close(done)
+				}()
+
+				skip := binary.BigEndian.Uint32(req.Payload)
+				size, err := windowSize(req.Payload[4+skip:])
+				if err != nil {
+					log.Printf("Failed to get window size: %v", err)
+					continue
+				}
+				resize <- size
+			case "window-change":
+				size, err := windowSize(req.Payload)
+				if err != nil {
+					log.Printf("Failed to get window size: %v", err)
+					continue
+				}
+				resize <- size
+			}
+		}
+	}
+}
+
+func newPTY(channel ssh.Channel, resize chan *pty.Winsize) {
```

Comment:
newX generally implies creating a new pointer to an X and returning it but this is actually running the whole PTY.  How about runPTY?

Reply:
done